lacework/resource_lacework_integration_aws_eks_audit_log.go

```diff
@@ -0,0 +1,237 @@
+	},
+	"retries": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Default:     5,
+		Description: "The number of attempts to create the external integration.",
+	},
+	"intg_guid": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"sns_arn": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The SNS ARN.",
+	},
+	"credentials": {
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Required:    true,
+		Description: "The credentials needed by the integration.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"role_arn": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"external_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	},
+	"is_org": {
+		Type:     schema.TypeBool,
+		Computed: true,
+	},
+	"created_or_updated_time": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"created_or_updated_by": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+func resourceLaceworkIntegrationAwsEksAuditLogCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework           = meta.(*api.Client)
+		retries            = d.Get("retries").(int)
+		awsEksAuditLogData = api.AwsEksAuditData{
+			SnsArn: d.Get("snsArn").(string),
+			Credentials: api.AwsEksAuditCredentials{
+				RoleArn:    d.Get("credentials.0.role_arn").(string),
+				ExternalID: d.Get("credentials.0.external_id").(string),
+			},
+		}
+	)
+
+	awsEksAuditLog := api.NewCloudAccount(d.Get("name").(string),
+		api.AwsEksAuditCloudAccount,
+		awsEksAuditLogData,
+	)
+
+	if !d.Get("enabled").(bool) {
+		awsEksAuditLog.Enabled = 0
+	}
+
+	return resource.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		retries--
+		log.Printf("[INFO] Creating %s cloud account integration\n", api.AwsEksAuditCloudAccount.String())
+		response, err := lacework.V2.CloudAccounts.Create(awsEksAuditLog)
+		if err != nil {
+			if retries <= 0 {
+				return resource.NonRetryableError(
+					fmt.Errorf("error creating %s cloud account integration: %s",
+						api.AwsEksAuditCloudAccount.String(), err,
+					))
+			}
+			log.Printf(
+				"[INFO] Unable to create %s cloud account integration. (retrying %d more time(s))\n%s\n",
+				api.AwsEksAuditCloudAccount.String(), retries, err,
+			)
+			return resource.RetryableError(fmt.Errorf(
+				"unable to create %s cloud account integration (retrying %d more time(s))",
+				api.AwsEksAuditCloudAccount.String(), retries,
+			))
+		}
+
+		cloudAccount := response.Data
+		d.SetId(cloudAccount.IntgGuid)
+		d.Set("name", cloudAccount.Name)
+		d.Set("intg_guid", cloudAccount.IntgGuid)
+		d.Set("enabled", cloudAccount.Enabled == 1)
+
+		d.Set("created_or_updated_time", cloudAccount.CreatedOrUpdatedTime)
+		d.Set("created_or_updated_by", cloudAccount.CreatedOrUpdatedBy)
+		d.Set("type_name", cloudAccount.Type) // @afiune should we deprecate?
```

Comment:
What's this deprecation comment about?

Reply:
This is actually a carryover from `resource_lacework_integration_aws_ct.go`.
I'm not really sure of the context for this.